sim/deathknight/talents_blood.go

```diff
@@ -33,7 +33,7 @@ func (dk *Deathknight) ApplyBloodTalents() {
 	dk.applyScentOfBlood()
 
 	// Two Handed Specialization
-	if dk.HasMHWeapon() && dk.Equip[proto.ItemSlot_ItemSlotMainHand].HandType == proto.HandType_HandTypeTwoHand {
+	if dk.MainHand().HandType == proto.HandType_HandTypeTwoHand {
```

Comment:
I'm pretty sure calls like this will panic (due to `MainHand()` returning `nil`) when there is no item equipped for the slot

Reply:
These just return a pointer to an Equipment slot ([17]Item), so they're never nil.